### PR TITLE
Add mergeCollection method to Onyx.update

### PR DESCRIPTION
**Pull request by Julesssss**
Enables the mergeCollection Onyx method. To be tested in the Web-E PR.

### Details
Simply enables the mergeCollection Onyx method.

### Related Issues
https://github.com/Expensify/Expensify/issues/214912

### Automated Tests

Added a new test specifically to test a mergeCollection Onyx update

### Linked PRs

- Web-E #1: https://github.com/Expensify/Web-Expensify/pull/34205
- Web-E #2: https://github.com/Expensify/Web-Expensify/pull/34144/files

### CODE REVIEW

**Comment by Julesssss** [comment]
I have read the CLA Document and I hereby sign the CLA

File: tests/unit/onyxTest.js [hunk 1]
```diff
@@ -358,6 +358,89 @@ describe('Onyx', () => {
...
+        const mockCallback = jest.fn(data => valuesReceived[data.ID] = data.value);
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_KEY,
+            initWithStoredValues: false,
+            callback: mockCallback,
+        });
+
+        return waitForPromisesToResolve()
+            .then(() => {
+                // GIVEN the initial Onyx state: {test_1: {existingData: 'test',},test_2: {existingData: 'test',}}
```
Comment (mountiny) [inline tests/unit/onyxTest.js:372]: NAB: OCD
```suggestion
                // GIVEN the initial Onyx state: {test_1: {existingData: 'test',}, test_2: {existingData: 'test',}}
```

**Review by mountiny — DISMISSED** [review]
One very small nitty picky comment, but otherwise LGTM

File: tests/unit/onyxTest.js (hunk 1 above)
Comment (Julesssss) [reply tests/unit/onyxTest.js]: Fixed. Now you can sleep happily tonight :D

**Review by mountiny — APPROVED** [review]
Thanks! Appreciate the change for better sleep 🙌 

Did you assign other reviewer on purpose or was it gh blip? should we wait for Daniel with merging?

**Review by danieldoglas — APPROVED** [review]
LGTM

**Comment by danieldoglas** [comment]
This failed on publish @mountiny @Julesssss

**Comment by Julesssss** [comment]
> This failed on publish @mountiny @Julesssss

Hey @danieldoglas. Sorry, I'm currently OOO due to Covid. What do you mean by failed on publish?

**Comment by mountiny** [comment]
@Julesssss Hope you having mild symptoms!

I think he means this workflow failed https://github.com/Expensify/react-native-onyx/actions/runs/2612339269

I am not sure if it is fine to just retry, it might be realted to some of the env changes made recently

**Comment by mountiny** [comment]
I will try and rerun it, _what could go wrong_

**Comment by Julesssss** [comment]
Thanks. That makes sense. 

I retried, but the action failed because the tag already existed. Now attempting to delete the tag and re-run the action.

**Pull request merged**